### PR TITLE
Add sphinx extension

### LINKED ISSUE #865: Bug(Sphinx): Autodoc fails to document wrapped functions
Hello, we are using Procrastinate for a work project and I'm having a difficult time getting wrapped functions to document with Sphinx correctly.

## Issue

When wrapping a function it fails to document.

For example 
```py
from somewhere import procrastinate_app as pa

@pa.periodic(cron="*/30 * * * *")  # type: ignore[misc]
@pa.task()  # type: ignore[misc]
async def sync_thing(**kwargs: Any) -> list[Thing] | Message:
    """Do thing

    Args:
        **kwargs: Keyword arguments passed to the task.
            These include ``timestamp`` from Procrastinate.

    Returns:
        Message: A message.
    """
    return await fetch_testboxes()
 ```
Will generate nothing for Sphinx's `autodoc`
```rst

API Reference
-------------

.. automodule:: path.to.module.tasks
    :members:
```

I think possibly related to https://github.com/sphinx-doc/sphinx/issues/3783, but I'm wondering if it can be fixed from your side.

### PR DESCRIPTION
**Pull request by ewjoachim**
Closes #865

Procrastinate Tasks don't get documented by Sphinx Autodoc. It's a bit sad, but with a very short sphinx extension, we can solve it. So... I guess we have a new contrib.

### Successful PR Checklist:

- [x] Tests
  - [ ] (not applicable?)
- [x] Documentation
  - [ ] (not applicable?)

#### PR label(s): 
  - [ ] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20breaking%20%F0%9F%92%A5
  - [ ] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20feature%20%E2%AD%90%EF%B8%8F
  - [x] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20bugfix%20%F0%9F%95%B5%EF%B8%8F
  - [ ] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20miscellaneous%20%F0%9F%91%BE
  - [ ] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20dependencies%20%F0%9F%A4%96
  - [ ] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20documentation%20%F0%9F%93%9A

### CODE REVIEW

File: procrastinate_demos/demo_async/tasks.py [hunk 1]
```diff
@@ -5,9 +5,11 @@
 
 @app.task(queue="sums")
 async def sum(a, b):
+    """Sum two numbers."""
     return a + b
 
 
 @app.task(queue="defer")
 async def defer():
+    """Defer a nother task."""
```
Comment (medihack) [inline procrastinate_demos/demo_async/tasks.py:14]: "another"

File: tests/conftest.py [hunk 2]
```diff
@@ -25,6 +25,10 @@
     if key.startswith("PROCRASTINATE_"):
         os.environ.pop(key)
 
+# Infortunately, we need the sphinx fixtures even though they generate an "app" fixture
```
Comment (medihack) [inline tests/conftest.py:28]: "Unfortunately"

**Comment by medihack** [comment]
Nice. Procrastinate is full of little details. Let's get it in there :-)

File: procrastinate_demos/demo_async/tasks.py (hunk 1 above)
Comment (ewjoachim) [reply procrastinate_demos/demo_async/tasks.py:14]:
```suggestion
    """Defer a another task."""
```

File: tests/conftest.py (hunk 2 above)
Comment (ewjoachim) [reply tests/conftest.py:28]:
```suggestion
# Unfortunately, we need the sphinx fixtures even though they generate an "app" fixture
```

**Pull request merged**